go/proof_data_test.go

```diff
@@ -54,7 +54,7 @@ func CheckLeafTestData(tb testing.TB) map[string]CheckLeafTestStruct {
 type CheckAgainstSpecTestStruct struct {
 	Proof *ExistenceProof
 	Spec  *ProofSpec
-	IsErr bool
+	Err   string
```

Comment:
should we make this `error` instead of `string`?

Reply:
maybe not worth doing this tbh. I see the testdata json structs are unmarshalled to this struct and we'd need to handle custom unmarshalJSON functionality to handle the built-in error interface. 

Using `string` is fine by me 👍🏻

Reply:
yup, I tried doing `error` but felt it was too much work to dig further into